client/MentorPanel/src/components/video.js

```diff
@@ -22,32 +22,52 @@ const FaveButton = () => {
     )
 }
 
-const Video = () => {
+const VideoPlayer = ({ width }) => {
     const dispatch = useDispatch()
     const isIdle = useSelector(state => state.isIdle)
     const mentor = useSelector(state => state.mentors_by_id[state.current_mentor])
-    var src = mentor ?
-        isIdle ? idleUrl(mentor) : videoUrl(mentor)
-        : ''
+    const src =
+        mentor ?
+            isIdle ? idleUrl(mentor) : videoUrl(mentor)
+            : ''
 
     const onEnded = () => {
         dispatch(setIdle())
     }
 
     return (
-        <div id='video-container'>
-            <ReactPlayer
-                url={src}
-                onEnded={onEnded}
-                width='100%'
-                loop={isIdle}
-                controls={true}
-                playing={true}
-                playsinline={true}
-                webkit-playsinline='true' />
-            <FaveButton />
-        </div>
+        <ReactPlayer
+            url={src}
+            onEnded={onEnded}
+            loop={isIdle}
+            width={width}
+            height={width * 0.895}
+            controls={true}
+            playing={true}
+            playsinline={true}
+            webkit-playsinline='true' />
     )
 }
 
+class Video extends React.Component {
```

Comment:
does this need to be a component class? 

I guess the q is whether you can use the `window` variable inside of `useEffect` and/or could we check for the existence of `window`, e.g

```
const maxWidth = 500
const windowWidth = typeof(window) !== undefined? window.width: maxWidth
const width = Math.min(windowWidth, windowWidth)
```

We should be consistent and use functional components by default/whereever practical

Reply:
If width is not a state or property that's updating, I don't think it would re-render at all is the issue.

Reply:
I can try the method where we check if window exists before calling it.

Reply:
ok let me know. if doesn't work then use the class is fine. Would just like to settle on using all one or the other--functional components or ES6-class components. Since React hooks require functional components, let's try go in that dir where ever possible

Reply:
updated